kotlinx-coroutines-core/native/src/Dispatchers.kt

```diff
@@ -19,6 +22,35 @@ public actual object Dispatchers {
     internal fun injectMain(dispatcher: MainCoroutineDispatcher) {
         injectedMainDispatcher = dispatcher
     }
+
+    internal val IO: CoroutineDispatcher = DefaultIoScheduler
 }
 
+internal object DefaultIoScheduler : CoroutineDispatcher() {
```

Comment:
on non Darwin platforms default dispatcher is backed by additional `newFixedThreadPoolContext(availableProcessors)` (comparing to global queue on Darwin) - does it make sense to use single pool to share threads for both IO and Default there? will it give something?

Reply:
>will it give something?

Nothing, really, unless the full-blown scheduler is implemented.
JVM implementation does that because it has a dedicated implementation that allows to avoid unnecessary context switching pro-actively (e.g. by tweaking its own scheduling policies). Having a single pool "shared" via `limitedParallelism` won't achieve that